pkg/remote/google_compute_scanner_test.go

```diff
@@ -148,3 +148,143 @@ func TestGoogleComputeFirewall(t *testing.T) {
+			dirName:  "google_compute_router_empty",
+			response: []*assetpb.ResourceSearchResult{},
+			wantErr:  nil,
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 0)
+			},
+		},
+		{
+			test:    "multiples compute routers",
+			dirName: "google_compute_router_multiple",
+			response: []*assetpb.ResourceSearchResult{
+				{
+					AssetType:   "compute.googleapis.com/Router",
+					DisplayName: "test-router-0",
+					Name:        "//compute.googleapis.com/projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-0",
+				},
+				{
+					AssetType:   "compute.googleapis.com/Router",
+					DisplayName: "test-router-1",
+					Name:        "//compute.googleapis.com/projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-1",
+				},
+				{
+					AssetType:   "compute.googleapis.com/Router",
+					DisplayName: "test-router-2",
+					Name:        "//compute.googleapis.com/projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-2",
+				},
+			},
+			wantErr: nil,
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 3)
+
+				assert.Equal(t, got[0].ResourceId(), "projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-0")
+				assert.Equal(t, got[0].ResourceType(), googleresource.GoogleComputeRouterResourceType)
+
+				assert.Equal(t, got[1].ResourceId(), "projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-1")
+				assert.Equal(t, got[1].ResourceType(), googleresource.GoogleComputeRouterResourceType)
+
+				assert.Equal(t, got[2].ResourceId(), "projects/cloudskiff-dev-raphael/regions/us-central1/routers/test-router-2")
+				assert.Equal(t, got[2].ResourceType(), googleresource.GoogleComputeRouterResourceType)
+			},
+		},
+		{
+			test:        "should return access denied error",
+			dirName:     "google_compute_router_empty",
+			wantErr:     nil,
+			responseErr: status.Error(codes.PermissionDenied, "The caller does not have permission"),
+			setupAlerterMock: func(alerter *mocks.AlerterInterface) {
+				alerter.On(
+					"SendAlert",
+					googleresource.GoogleComputeRouterResourceType,
+					alerts.NewRemoteAccessDeniedAlert(
+						common.RemoteGoogleTerraform,
+						remoteerr.NewResourceListingError(
+							status.Error(codes.PermissionDenied, "The caller does not have permission"),
+							googleresource.GoogleComputeRouterResourceType,
+						),
+						alerts.EnumerationPhase,
+					),
+				).Once()
+			},
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 0)
+			},
+		},
+	}
+
+	providerVersion := "3.78.0"
+	schemaRepository := testresource.InitFakeSchemaRepository("google", providerVersion)
+	googleresource.InitResourcesMetadata(schemaRepository)
+	factory := terraform.NewTerraformResourceFactory(schemaRepository)
+
+	for _, c := range cases {
+		t.Run(c.test, func(tt *testing.T) {
+			shouldUpdate := c.dirName == *goldenfile.Update
```

Comment:
You could simplify this test a lot since we are not in deep mode, you can remove dirName and other deep mode related stuff. Check my PR on `google_compute_instance` for example